client/components/theme/index.jsx

```diff
@@ -233,8 +243,49 @@ export class Theme extends Component {
 		);
 	};
 
+	goToCheckout = () => {
+		const { siteSlug, eligibleForProPlan } = this.props;
+		const plan = eligibleForProPlan && isEnabled( 'plans/pro-plan' ) ? 'pro' : 'premium';
+
+		if ( siteSlug ) {
+			const params = new URLSearchParams();
+			params.append( 'redirect_to', window.location.href.replace( window.location.origin, '' ) );
+
+			window.location.href = `/checkout/${ encodeURIComponent(
+				siteSlug
+			) }/${ plan }?${ params.toString() }`;
+		}
+	};
+
+	getUpsellMessage() {
+		const { isPremiumThemesAvaiable, eligibleForProPlan, theme, price, translate } = this.props;
+		const hasPrice = /\d/g.test( price );
+
+		if ( ! hasPrice && theme.price && ! isPremiumThemesAvaiable ) {
+			return translate( 'You have purchased an annual subscription for this theme' );
+		} else if ( isPremiumThemesAvaiable ) {
+			return translate( 'The premium theme is included in your plan.' );
+		}
+
+		return createInterpolateElement(
+			sprintf(
+				/* translators: the "price" is the price of the theme, example: U$50; the "planName" can be "Pro plan" or "Premium plan". */
+				translate(
+					'This premium theme is included in the <Link>%(planName)s</Link>, or you can purchase individually for %(price)s a year'
```

Comment:
I would omit the part about purchasing the theme individually; I know it's part of the mock-ups, but the intention is to stop selling the themes individually when this new UI launches. 

/cc @SaxonF Should this say something about premium theme subscriptions instead?

Reply:
is "or you can purchase a theme subscription for xx a year" a little clearer?

Reply:
I think that might actually muddy the waters since "a theme subscription" could be interpreted as a subscription that unlocks all themes.

How about, new sentence, "This theme can also be purchased individually for x per year."?

@sixhours I feel like I'm missing something in your point about "the intention is to stop selling the themes individually" - what do you mean by that?

Reply:
Let's leave this text as it is right now, to facilitate getting this out the door.